From 512bdd8f480c0790797521884145e3e6d8661254 Mon Sep 17 00:00:00 2001
From: Renato Araujo Oliveira Filho <renato.filho@canonical.com>
Date: Mon, 14 Sep 2015 17:38:48 -0300
Subject: [PATCH 2/3] Fixed 'FUNCTION_CALL_TRACE' macro.

Make sure that the macro print the correct elapsed time during the debug.
---
 libbuteosyncfw/common/LogMacros.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libbuteosyncfw/common/LogMacros.h b/libbuteosyncfw/common/LogMacros.h
index bac9ef6..90f2c6f 100644
--- a/libbuteosyncfw/common/LogMacros.h
+++ b/libbuteosyncfw/common/LogMacros.h
@@ -29,6 +29,7 @@
 #include <QTime>
 #include <QDebug>
 #include <QDateTime>
+#include <QScopedPointer>
 #include "Logger.h"
 
 //! Helper macro for writing log messages. Avoid using directly.
@@ -52,7 +53,9 @@
   * Creates a trace message to log when the function is entered and exited.
   * Logs also to time spent in the function.
   */
-# define FUNCTION_CALL_TRACE if (Buteo::Logger::instance()->getLogLevel() >= 9) Buteo::LogTimer timerDebugVariable(QString(__PRETTY_FUNCTION__));
+# define FUNCTION_CALL_TRACE QScopedPointer<Buteo::LogTimer> timerDebugVariable; \
+    if (Buteo::Logger::instance()->getLogLevel() >= 9) \
+        timerDebugVariable.reset(new Buteo::LogTimer(QString(__PRETTY_FUNCTION__)));
 
 namespace Buteo {
 
-- 
2.1.4

